File: apps/desktop/src/types/task.ts
export interface Task {
    id: string;
    agent: string;
    description: string;
    status: "pending" | "running" | "completed" | "failed" | "cancelled";
    progress: number;
    dependencies: string[];
    priority: number;
    error_message?: string;
}

export interface TaskPlan {
    plan_id: string;
    tasks: Task[];
    estimated_time_seconds: number;
}

File: core/models/task.py
from __future__ import annotations

from pydantic import BaseModel


class Task(BaseModel):
    id: str
    type: str
    description: str
    agent: str | None = None
    dependencies: list[str] = []
    priority: int = 5
    status: str = "pending"
    input_data: dict = {}
    output_data: dict | None = None
    max_retries: int = 3
    timeout_seconds: int = 300
    ram_required_mb: int = 50
    error_message: str | None = None
